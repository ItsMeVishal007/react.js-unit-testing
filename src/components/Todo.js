import React from 'react'

const Todo = ({todo}) => {
  const {id, title, complete} = todo;
  const h1 = <h1>{title}</h1>
  const text = complete ? <strike>{h1}</strike> : h1;
  return (
    <div data-testid={`todo-${id}`}>{text}</div>
  )
}

export default Todo;
