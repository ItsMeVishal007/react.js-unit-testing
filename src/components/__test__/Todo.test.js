import { screen, cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer'
import Todo from '../Todo';

afterEach(()=>{
  cleanup();
})

test("todo item should be false =>", () => {
  const todo = { id: 0, title: 'wash dishes', complete: false };
  render(<Todo todo={todo}/>);
  const todoElement = screen.getByTestId('todo-0');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('wash dishes');
})

test("todo item should be true =>", () => {
  const todo = { id: 1, title: 'wash car', complete: true };
  render(<Todo todo={todo}/>);
  const todoElement = screen.getByTestId('todo-1');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('wash car');
})

test("matches snapshot", ()=> {
  const todo = { id: 1, title: 'wash car', complete: true };
  const tree = renderer.create(<Todo todo={todo} />);
  expect(tree).toMatchSnapshot();
})