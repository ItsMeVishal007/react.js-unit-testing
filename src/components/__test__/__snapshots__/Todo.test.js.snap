// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h1>
      wash car
    </h1>
  </strike>
</div>
`;
